src/utils/index.tsx

```diff
@@ -0,0 +1,45 @@
+/*!
+ * Copyright (c) 2017-Present, Okta, Inc. and/or its affiliates. All rights reserved.
+ * The Okta software accompanied by this notice is provided pursuant to the Apache License, Version 2.0 (the "License.")
+ *
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0.
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *
+ * See the License for the specific language governing permissions and limitations under the License.
+ */
+
+import { toRelativeUrl, OktaAuth, AuthState } from '@okta/okta-auth-js';
+
+export const getRelativeUri = (originalUri: string): string => {
+  if (!originalUri) {
+    return '/';
+  }
+  let uri = originalUri;
+  if (uri.startsWith(window.location.origin)) {
+    uri = toRelativeUrl(uri, window.location.origin);
+  }
+  if (uri.startsWith('/#')) {
+    // strip the lead '/#' from the uri
+    uri = uri.slice(2);
+  }
+  return uri;
+};
+
+export const waitForAuthenticated = (
+  oktaAuth: OktaAuth
+): Promise<boolean> => {
+  return new Promise((resolve) => {
```

Comment:
I think it makes more sense to use `await oktaAuth.isAuthenticated()` here, rather than the authState

Reply:
`oktaAuth.isAuthenticated()` can't be used for this purpose (purpose is to use some waiter util in react-router [`loader`](https://reactrouter.com/en/main/route/loader)).
`oktaAuth.isAuthenticated()` will return promise that will immediately be resolved with true/false rather that actually waiting for auth state with `isAuthenticated: true`

Reply:
> oktaAuth.isAuthenticated() will return promise that will immediately be resolved with true/false rather that actually waiting for auth state with isAuthenticated: true

I'm not sure I follow this. `oktaAuth.isAuthenticated` will attempt to read the tokens from the tokenManager and determine their validity (and attempt a renew) before resolving